src/ngAnimate/animateQueue.js

```diff
@@ -0,0 +1,558 @@
+            // we check for null directly in the event that the application already called
+            // .enabled() with whatever arguments that it provided it with
+            if (animationsEnabled === null) {
+              animationsEnabled = true;
+            }
+          });
+        });
+      }
+    );
+
+    var callbackRegistry = {};
+    var callbackCounter = 0;
+
+    // remember that the classNameFilter is set during the provider/config
+    // stage therefore we can optimize here and setup a helper function
+    var classNameFilter = $animateProvider.classNameFilter();
+    var isAnimatableClassName = !classNameFilter
+              ? function() { return true; }
+              : function(className) {
+                return classNameFilter.test(className);
+              };
+
+    function nextCallbackID() {
+      return '__animateID__' + (callbackCounter++);
+    }
+
+    function findCallbacks(element, event) {
+      // TODO use parentNode.contains
+      var isGeneratedID = false;
+      var id = element.attr('id');
+      if (!element.attr('id')) {
+        isGeneratedID = true;
+        id = nextCallbackID();
+        element.attr('id', id);
+      }
+
+      var matches = [];
+      var entries = callbackRegistry[event];
+      if (entries) {
+        forEach(entries, function(entry) {
+          var node = entry.node;
+          if (node.getAttribute('id') === id || node.querySelector('#' + id)) {
+            matches.push(entry.callback);
+          }
+        });
+      }
+
+      if (isGeneratedID) {
+        element.removeAttr('id');
+      }
+
+      return matches;
+    }
+
+    // $animate.on('enter', element, function() {
+    //
+    // });
+    // $animate.on('enter', '[ng-view]', function() {
+    //
+    // });
+    // element.on('$animate:enter', j
+    function triggerCallback(event, element, phase, data) {
+      $$rAF(function() {
+        forEach(findCallbacks(element, event), function(callback) {
+          callback(element, phase, data);
+        });
+      });
+    }
+
+    return {
+      on: function(event, container, callback) {
+        var node = extractElementNode(container);
+        callbackRegistry[event] = callbackRegistry[event] || [];
+        callbackRegistry[event].push({
+          node: node,
+          callback: callback
+        });
+      },
+
+      off: function(event, container, callback) {
+        var entries = callbackRegistry[event];
+        if (!entries) return;
+
+        switch (arguments.length) {
+          case 1:
+            callbackRegistry[event] = null;
+            break;
+
+          case 2:
+            callbackRegistry[event] = filterFromRegistry(entries, container);
+            break;
+
+          case 3:
+            callbackRegistry[event] = filterFromRegistry(entries, container, callback);
+            break;
+        }
+
+        function filterFromRegistry(list, matchContainer, matchCallback) {
+          var containerNode = matchContainer[0];
```

Comment:
You are using `extractElementNode(container)` for registering. Shouldn't the same be used for unregistering ?